src/commissaire_service/storage/storehandlermanager.py

```diff
@@ -122,46 +127,57 @@ def list_container_managers(self, cluster_type=None):
                         container_manager = container_manager_class(config)
                         self._container_managers.append(container_manager)
                     else:
-                        logger = self._get_logger()
-                        logger.warn(
+                        self._logger.warn(
                             'A container manager is already established, '
                             'skipping {} as configured for store handler '
                             '"{}"'.format(
                                 container_manager_class.__name__,
                                 handler_type.__name__))
 
-        if cluster_type:
-            result = [x for x in self._container_managers
-                      if x.cluster_type == cluster_type]
-        else:
-            result = list(self._container_managers)
+        return self._container_managers
 
-        return result
+    def _create_handler(self, definition):
+        """
+        Creates a handler instance from a handler definition, and add the
+        handler instance to various internal data structures.
+        """
+        handler_type, config, model_types = definition
+        handler = handler_type(config)
+        self._handlers_by_name[config['name']] = handler
+        new_items = {mt: handler for mt in model_types}
+        self._handlers_by_model_type.update(new_items)
+        return handler
 
     def _get_handler(self, model):
         """
         Looks up, and if necessary instantiates, a StoreHandler instance
         for the given model.  Raises KeyError if no handler is registered
         for that type of model.
         """
-        handler = self._handlers.get(type(model))
+        handler = None
+        model_type = type(model)
+
+        # Special case: If the model is a Host, check for a handler name
+        #               in its "source" attribute and use the definition
+        #               registered under that name.
+        if model_type is Host:
```

Comment:
No issue here, but making a note that over time more models will probably have this special case.